nixos/modules/services/monitoring/alertmanager-executor.nix

```diff
@@ -0,0 +1,70 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alertmanager-executor;
+  settingsFormat = pkgs.formats.yaml {};
+
+in {
+  options = {
+    services.alertmanager-executor = {
+      enable = mkEnableOption "alertmanager-executor (aka prometheus-am-executor)";
+
+      user = mkOption {
+        type = types.str;
+        default = "alertmanager-executor";
+        description = lib.mdDoc "User to run the service and executed commands under.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "alertmanager-executor";
+        description = lib.mdDoc "Group to run the service and executed commands under.";
+      };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = {};
+        example = {
+          listen_address = "localhost:23222";
+          commands = [
+            {
+              cmd = "echo";
+              args = [ "alert" "firing" ];
+            }
+          ];
+        };
+        description = lib.mdDoc ''
+          Configuration for alertmanager-executor. Documentation:
+          <https://github.com/imgix/prometheus-am-executor#configuration-file-format>.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users = optionalAttrs (cfg.user == "alertmanager-executor") {
+      alertmanager-executor = {
+        description = "Execute command based on Prometheus alerts";
+        isSystemUser = true;
+        group = cfg.group;
+      };
+    };
+
+    users.groups = optionalAttrs (cfg.group == "alertmanager-executor") {
+      alertmanager-executor = {};
+    };
+
+    systemd.services.alertmanager-executor = {
+      description = "Execute command based on Prometheus alerts";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.alertmanager-executor}/bin/prometheus-am-executor -v -f ${settingsFormat.generate "executor.yaml" cfg.settings}";
```

Comment:
Could you also please add an option to append command-line arguments.

Reply:
Fixed in [b8e9eac](https://github.com/NixOS/nixpkgs/pull/189119/commits/b8e9eacbe462ea1aa3ea1c492581c1d529f8c8f3)